frontend/src/components/project_page/pages/ProjectLandingPage/components/StarButton.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { FC, useEffect, useState } from "react";
+import { useSession } from "../../../../../hooks/session";
+import { Project, Member } from "../../../../../shared/Interfaces";
+import LoadingSpinner from "../../../../Spinners/LoadingSpinner";
+interface Props {
+  project: Project;
+}
+const StarButton: FC<Props> = ({ project }) => {
+  const { loading, isLoggedIn, user } = useSession();
+  const [starButton, setStarButton] = useState(false);
+  const [unstarButton, setUnstarButton] = useState(false);
+  const [starrers, setStarrers] = useState<string[]>([]);
+  const [projectOwner, setProjectOwner] = useState(false);
+
+  useEffect(() => {
+    fetch(`/api/v1/projects/${project.id}`).then(async (response) => {
+      if (response.status === 200) {
+        const data = await response.json();
+        setStarrers(data.starrers);
+
+        if (data.members) {
+          data.members.map((m: Member) => {
+            if (
+              m.roleName === "owner" &&
```

Comment:
I feel like it's actually okay to allow an owner to star their own project (just like GitHub lets you star your own repo)